src/secw_security_wallet_server.cc

```diff
@@ -0,0 +1,722 @@
+*/
+
+/*
+@header
+    secw_security_wallet_server - secw broker agent
+@discuss
+@end
+*/
+
+#include "fty_security_wallet_classes.h"
+
+#include <sstream>
+#include <cxxtools/jsonserializer.h>
+
+#include "secw_exception.h"
+#include "secw_configuration.h"
+#include "secw_document.h"
+
+#include "secw_helpers.h"
+
+using namespace secw;
+using namespace std::placeholders;
+
+SecurityWalletServer::SecurityWalletServer( const std::string & configurationPath,
+                                            const std::string & databasePath,
+                                            FctNotifier notifier)
+    :   m_activeWallet(configurationPath, databasePath),
+        m_notifier(notifier)
+{
+    //initiate the commands handlers
+    m_supportedCommands[GET_PORTFOLIO_LIST] = std::bind(&SecurityWalletServer::handleGetListPortfolio, this, _1, _2);
+
+    m_supportedCommands[GET_CONSUMER_USAGES] = std::bind(&SecurityWalletServer::handleGetConsumerUsages, this, _1, _2);
+    m_supportedCommands[GET_PRODUCER_USAGES] = std::bind(&SecurityWalletServer::handleGetProducerUsages, this, _1, _2);
+
+    m_supportedCommands[GET_LIST_WITH_SECRET] = std::bind(&SecurityWalletServer::handleGetListDocumentsWithSecret, this, _1, _2);
+    m_supportedCommands[GET_LIST_WITHOUT_SECRET] = std::bind(&SecurityWalletServer::handleGetListDocumentsWithoutSecret, this, _1, _2);
+
+    m_supportedCommands[GET_WITHOUT_SECRET] = std::bind(&SecurityWalletServer::handleGetDocumentWithoutSecret, this, _1, _2);
+    m_supportedCommands[GET_WITH_SECRET] = std::bind(&SecurityWalletServer::handleGetDocumentWithSecret, this, _1, _2);
+
+    m_supportedCommands[GET_WITHOUT_SECRET_BY_NAME] = std::bind(&SecurityWalletServer::handleGetDocumentWithoutSecretByName, this, _1, _2);
+    m_supportedCommands[GET_WITH_SECRET_BY_NAME] = std::bind(&SecurityWalletServer::handleGetDocumentWithSecretByName, this, _1, _2);
+    
+    m_supportedCommands[CREATE] = std::bind(&SecurityWalletServer::handleCreate, this, _1, _2);
+    m_supportedCommands[DELETE] = std::bind(&SecurityWalletServer::handleDelete, this, _1, _2);
+    m_supportedCommands[UPDATE] = std::bind(&SecurityWalletServer::handleUpdate, this, _1, _2);
+}
+
+std::vector<std::string> SecurityWalletServer::handleRequest(const Sender & sender, const std::vector<std::string> & payload)
+{
+    try
+    {
+        if(payload.size() == 0)
+        {
+            throw SecwProtocolErrorException("Command frame is empty");
+        }
+        
+        Command cmd = payload.at(0);
+        
+        if(cmd == "ERROR" || cmd == "OK")
+        {
+            //avoid loop
+            return {};
+        }
+        
+        //check if the command exist in the system
+        if( m_supportedCommands.count(cmd) == 0)
+        {
+            throw SecwUnsupportedCommandException(cmd + " not supported");
+        }
+
+        FctCommandHandler cmdHandler = m_supportedCommands[cmd];
+        
+        // Declaring new vector 
```

Comment:
It can be written like this:
 std::vector<std::string> params(payload.begin() + 1, payload.end());